Celeste.Mod.mm/Mod/UI/TextMenuExt.cs

```diff
@@ -1353,5 +1353,93 @@ public class BatchModeContext : IDisposable {
             }
         }
 
+        public class SubMenuWithInputs : TextMenu.Item {
```

Comment:
Might be worth to try and implement a `EaseInDecorator` in order to reduce code repetition.
something like this:

``` c#
public class EaseInDecorator<T> : TextMenu.Item, IItemExt where T : TextMenu.Item, IItemExt {
            private T inner;

            public EaseInDecorator(T inner) {
                this.inner = inner;
            }

}
```